### PR TITLE
Added tests for Radio

### LINKED ISSUE #966: Add tests for Radio

### PR DESCRIPTION
**Pull request by vysakh-m**
Fixes #966

### CODE REVIEW

**Comment by vysakh-m** [comment]
@amaldinesh7 _a Please review

File: tests/Radio.test.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+  it("should call onChange when radio value is changed", () => {
+    const onChange = jest.fn();
+    const { getByText } = render(
+      <Radio label="Radio" onChange={onChange}>
+        <Radio.Item label="option1" name="options" value="option1" />
+      </Radio>
+    );
+    userEvent.click(getByText("option1"));
+    expect(onChange).toHaveBeenCalledTimes(1);
```
Comment (amaldinesh7) [inline tests/Radio.test.js:39]: @ajmaln Do we have to test what value is getting returned on onChange here? Can we test that? (Doubt)

**Review by amaldinesh7 — APPROVED** [review]

**Comment by amaldinesh7** [comment]
@ajmaln _a Please review

File: tests/Radio.test.js (hunk 1 above)
Comment (ajmaln) [reply tests/Radio.test.js:39]: @amaldinesh7 the arguments can be tested through `expect(onChange).toHaveBeenCalledWith()`. However, the `target` property will be null for the events due to synthetic event nullification. 

I found a way by including the assertion in the mock function itself.

```js
it("should call onChange when radio value is changed", () => {
    const onChange = jest.fn((event) => {
      expect(event.target.value).toBe("option1");
    });
    const { getByText } = render(
      <Radio label="Radio" onChange={onChange}>
        <Radio.Item label="option1" name="options" value="option1" />
      </Radio>
    );
    userEvent.click(getByText("option1"));
    expect(onChange).toHaveBeenCalledTimes(1);
});
```

This seems to be working as expected and fails when the assertion value is wrong. Let's do this for testing values.

**Comment by vysakh-m** [comment]
@ajmaln As suggested, I added the test for uncontrolled radio inputs to test if they are getting checked when clicked.
```
it("should be checked on clicking the radio item", () => {
    const { getByRole } = render(
      <Radio label="Radio">
        <Radio.Item label="option1" name="options" value="option1" />
      </Radio>
    );
    const radio = getByRole("radio");
    userEvent.click(radio);
    expect(radio).toBeChecked();
  });
```
But this test was failing & returned `Received element is not checked`. When I checked the behaviour of uncontrolled radio inputs in Storybook, I found that radio buttons are not getting checked when they are clicked.

cc @amaldinesh7

**Comment by vysakh-m** [comment]
@ajmaln _a Resolved suggestions. Also, I've added the fix for uncontrolled radio input. Please review.

**Review by ajmaln — APPROVED** [review]

**Pull request merged**